pkg/volume/csi/csi_client.go

```diff
@@ -162,8 +162,11 @@ func (c *csiDriverClient) NodeGetInfo(ctx context.Context) (
 			return true, nil
 		}
 		// kubelet plugin registration service not implemented is a terminal error, no need to retry
-		if strings.Contains(getNodeInfoError.Error(), "no handler registered for plugin type") {
-			return false, getNodeInfoError
+		if getNodeInfoError != nil {
+			if strings.Contains(getNodeInfoError.Error(), "no handler registered for plugin type") {
```

Comment:
Thanks for pointing out that this error gets generated from pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go.

However, this getNodeInfoError is from nodeGetInfoV1() which I shouldn't go through that path. So I think this check is unnecessary.

Reply:
This check does look unnecessary to me. It doesn't look like anything inside `nodeGetInfoV1` would return that error. I can't recall why it was added. @taragu do you remember?

Reply:
Yeah I agree this is unnecessary since that error doesn't come from `nodeGetInfoV1`. I think it was added by mistake and can be safely removed.